### PR TITLE
Remove redundant `% size` from queue APIs

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
Closes https://github.com/au-ts/sddf/issues/111.

Brought up by @isubasinghe.

### CODE REVIEW

**Review by isubasinghe — DISMISSED** [review]
Thanks for making this change!

File: include/sddf/network/queue.h [hunk 1]
```diff
@@ -85,7 +85,7 @@ static inline bool net_queue_empty_active(net_queue_handle_t *queue)
  */
 static inline bool net_queue_full_free(net_queue_handle_t *queue)
 {
-    return !((queue->free->tail + 1 - queue->free->head) % queue->size);
+    return !(queue->free->tail + 1 - queue->free->head);
```
Comment (Courtney3141) [inline include/sddf/network/queue.h:88]: Both these changes to the full conditions are incorrect. 

Firstly, remember that I haven't yet corrected the issue raised before that if the queue size is `S`, we currently only use `S-1` entries. (This is next on my list of issues to fix.) 

Bearing this in mind, the idea with this function is to catch the case when the `tail` is `S-1` greater than the `head`. E.g.
`tail = head + S - 1`
Re-arranging this we get:
`tail + 1 - head = S`
which satisfies the old condition of:
`!((tail + 1 - head) % S)`

What you have changed it to instead is:
`tail + 1 - head != 0`, or `tail + 1 != head`
which means that a queue is only considered full when the `tail` is `UINT16_MAX - 1` greater than the `head` (much larger than most of our queues :-) ).

I believe the correct condition should be
`tail + 1 - head = size`

**Review by Courtney3141 — CHANGES_REQUESTED** [review]
The issue I discussed in detail in the net queue file applies to the other queue files as well.

File: include/sddf/network/queue.h (hunk 1 above)
Comment (Ivan-Velickovic) [reply include/sddf/network/queue.h:88]: Yes, my bad. I will fix that.

File: include/sddf/blk/queue.h [hunk 2]
```diff
@@ -132,7 +132,7 @@ static inline bool blk_resp_queue_empty(blk_queue_handle_t *h)
  */
 static inline bool blk_req_queue_full(blk_queue_handle_t *h)
 {
-    return !((h->req_queue->head - h->req_queue->tail + 1) % h->queue_size);
+    return h->req_queue->head - h->req_queue->tail + 1 == h->req_queue->size;
```
Comment (Courtney3141) [inline include/sddf/blk/queue.h:135]: I assume that blk seems to use head/tail in the opposite way to i2c/network/serial?

**Review by Courtney3141 — APPROVED** [review]

File: include/sddf/blk/queue.h (hunk 2 above)
Comment (Ivan-Velickovic) [reply include/sddf/blk/queue.h:135]: Yes, I noticed that and made a comment https://github.com/au-ts/sddf/issues/78#issuecomment-2109296110.

Comment (Ivan-Velickovic) [reply include/sddf/blk/queue.h:135]: Maybe @erichchan999 can fix that or I can make another PR later.

**Comment by Ivan-Velickovic** [comment]
I'll do a benchmark run tomorrow morning and then merge. Thanks for noticing my mistake @Courtney3141.

File: include/sddf/blk/queue.h (hunk 2 above)
Comment (erichchan999) [reply include/sddf/blk/queue.h:135]: Yep happy to make a separate PR once this is merged

**Comment by Courtney3141** [comment]
Since you will be away for a while, I re-ran a benchmark for you to check that performance is preserved and everything works as usual. I will merge this now.

**Pull request merged**